src/storage/singleton.rs

```diff
@@ -75,12 +75,26 @@ impl<'a, K, V> Iterator for IterMut<'a, K, V> {
     }
 }
 
+pub struct IntoIter<K, V> {
+    value: Option<(K, V)>,
```

Comment:
I think the `IntoIter` can be made smaller (even though `K` is basically always going to be a ZST :sweat_smile:) by producing `K` in the iterator instead. But that's a minor nit.